unittest.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Jun 26 13:53:41 2017
+
+@author: siddhant
+"""
+
+import unittest
+from point import Point
+import constants as c
+from line import Line
+
+
+p1 = Point(1.0, 2.0, 3.0)
+p2 = Point(2.0, 4.0, 6.0)
+l1 = Line(p1, p2)
+p3 = Point(1.0, 2.0, 3.0)
+p4 = Point(6.0, 12.0, 18.0)
+l2 = Line(p3, p4)
+
+class PointTestCase (unittest.TestCase):
+   
+   
+    #check if point is equal to given point (__eq__)
+    def test_point_equal(self):
+        self.assertEqual(p1, Point(1.0, 2.0, 3.0))
+     
+    #check points are not equal (__ne__)   
+    def test_point_notequal(self):
+        self.assertFalse(p1 == Point(1.0, 2.0, 8.0))
+    
+    #check if point negative is correct (__neg_)
+    def test_point_negative(self):
+        self.assertTrue(-p1 == Point(-1.0, -2.0, 3.0))
+        
+    #compare two points to check lesser (__lt__)
+    def test_point_lesser(self):
+        self.assertTrue(p1 < Point(2.0, 4.0, 6.0))
+     
+    #compare two points to check greater (__gt__)   
+    def test_point_greater(self):
+        self.assertFalse(p1 > Point(1.0, 2.0, 6.0))
+        
+    #check distance between two points (__sub__)  
+    def test_point_sub(self):
+        self.assertAlmostEqual(p2-p1, 3.74, places = 2)
+        
+    #check if x-coordinate matches    
+    def test_initial_x_property(self):
+        self.assertTrue(p1.x == 1.0)
+        
+    #check if y-coordinate matches    
+    def test_initial_y_propert(self):
+        self.assertTrue(p2.y == 4.0)
+        
+    #check if z-coordinate matches    
+    def test_initial_z_property(self):
+        self.assertTrue(p1.z == 3.0)
+    
+    #check if correct coordinate is returned    
+    def test_get_item(self):
```

Comment:
I was able to have the test work when slicing for one element. However, it threw up an error when slicing for two or more.